agent/src/beerocks/slave/agent_db.h

```diff
@@ -72,8 +75,165 @@ class AgentDB {
+            };
+            std::array<sBssid, eBeeRocksIfaceIds::IFACE_TOTAL_VAPS> bssids;
+        } front;
+
+        struct sBack {
+            explicit sBack(const std::string &iface_name_) : iface_name(iface_name_) {}
+            std::string iface_name;
+            sMacAddr iface_mac;
+        } back;
+
+        struct sClient {
+            sClient(sMacAddr bssid_, size_t association_frame_length_, uint8_t *association_frame_)
+                : bssid(bssid_), association_time(std::chrono::steady_clock::now()),
+                  association_frame_length(association_frame_length_)
+            {
+                std::copy_n(association_frame_, association_frame_length_,
+                            association_frame.begin());
+            }
+            sMacAddr bssid;
+            std::chrono::steady_clock::time_point association_time;
+            size_t association_frame_length;
+            std::array<uint8_t, ASSOCIATION_FRAME_SIZE> association_frame;
+        };
+        // Associated clients grouped by Client MAC.
+        std::unordered_map<sMacAddr, sClient> associated_clients;
+    };
+
+    /**
+     * @brief Get pointer to of the radio data struct of a specific interface. The function can
+     * accept either front or back iface name.
+     * 
+     * @param iface_name Interface name of a radio, front or back.
+     * @return std::unique_ptr<sRadio> to the radio struct if exist. Otherwise, nullptr.
+     */
+    sRadio *radio(const std::string &iface_name);
+
+    /**
+     * @brief Add radio node to the database. At least one of the input argument must not be empty.
+     * This function should be called only once with valid arguments. If called once and then called
+     * again, the radio struct on the database will not be updated with new arguments, and the
+     * function will return false.
+     * 
+     * @param front_iface_name Front iface name.
+     * @param back_iface_name Back iface name.
+     * @return true if a radio struct has been added to the database, otherwise return false.
+     */
+    bool add_radio(const std::string &front_iface_name, const std::string &back_iface_name);
+
+    /**
+     * @brief Get list of all radio objects on the database.
+     * This function shall be used in order to iterate over all the radios. 
+     * 
+     * @return const std::vector<std::string &>& Intefaces names list.
+     */
+    const std::vector<sRadio *> &get_radios_list() { return m_radios_list; };
+
+    /* Helper enum for get_mac_parent_radio() function */
+    enum class eMacType : uint8_t { ALL, RADIO, BSSID, CLIENT };
+
+    /**
+     * @brief Get a pointer to the parent radio struct of given MAC address. 
+     * 
+     * @param mac MAC address of radio interface or bssid.
+     * @return sRadio* A pointer to the radio struct containing the given MAC address.
+     */
+    sRadio *get_mac_parent_radio(const sMacAddr &mac, eMacType mac_type_hint = eMacType::ALL);
```

Comment:
```suggestion
    sRadio *get_radio_by_mac(const sMacAddr &mac, eMacType mac_type_hint = eMacType::ALL);
```
Event better, create a different method for each type of MAC address with a more descriptive name and get rid of last parameter

Reply:
Or at least add the last parameter to the doxygen comment

Reply:
Changed the function name, but I think it is better to use one instead of a few - once for each type.
Fixed the Doxygen documentation.